controllers/drplacementcontrol_controller.go

```diff
@@ -942,7 +946,7 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 
 	// Make sure we record the state that we are deploying
 	d.setDRState(rmn.Deploying)
-	setMetricsTimerFromDRState(rmn.Deploying, timerStart)
+	d.setMetricsTimerFromDRState(rmn.Deploying, timerStart)
```

Comment:
This will end up in the default case right, which is not what you want, I think?

Reply:
Also what happens to the timer if we return prematurely.  IOW, If we return in line 960?  before reaching line 971.

Reply:
I updated some things based on this feedback:

1. There's one timer per DRPC instance that all metrics can use independently. Assuming the timer has started in the `Deploying` state, it will continue to run until it reaches the `Deployed` state. Invalid state transitions (e.g. `Deploying` -> `Failingover`) will stop the timer correctly, but that's it. 
2. If `setMetricsTimerFromDRState` is called multiple times on the same state, the timer will start, but then ignore subsequent calls until the end state is reached. The logic for this is in `setMetricsTimer()`.